manifests/server/config.pp

```diff
@@ -29,16 +29,35 @@
     mode   => '0755',
   }
 
-  file { $cfg_file:
-    ensure  => present,
-    owner   => $owner,
-    group   => $group,
-    mode    => '0644',
-    require => [
-      File[$cfg_dir],
-      Class['dns::server::install']
-    ],
-    notify  => Class['dns::server::service'],
+  case $::osfamily {
+    'RedHat': {
+      file { $cfg_file:
+        ensure  => present,
+        owner   => $owner,
+        group   => $group,
+        mode    => '0644',
+        content => template("${module_name}/named.conf.erb"),
```

Comment:
The [debian `named.conf` file](https://wiki.debian.org/Bind9#named.conf_File) does not need to be replaced - it already works.  Mr. Crowley's redhat named.conf is basically implementing the same configuration as the debian version.